modules/calendar/add.php

```diff
@@ -0,0 +1,548 @@
+    $args['form_comments'] = "";
+    $args['event_date'] = $event_date;
+    $args['form_enddate'] = $event_end_date;
+    $args['duration'] = 0;  // no duration in provider events
+    $args['form_repeat'] = "1";  // Repeats box checked
+    $args['recurrspec'] = $recurrspec;
+    $args['starttime'] = $starttime;
+    $args['endtime'] = $endtime;
+    $args['form_allday'] = 0;
+    $args['form_apptstatus'] = "-";  // None
+    $args['form_prefcat'] = "0";  // None
+    $args['locationspec'] = $locationspec;
+
+    $def_facility = sqlFetchArray(sqlStatement( "SELECT u.facility_id
+                                                 FROM users u
+                                                 LEFT JOIN facility f on (u.facility_id = f.id)
+                                                 WHERE u.id = ?", array($_SESSION['authId']) ));
+    // default_user_facility denotes id of default facility of logged in user
+    $default_user_facility = $def_facility['facility_id'];
+    // Taking facility & billing facility as default facility of logged in user
+    $args['facility'] = $default_user_facility;
+    $args['billing_facility'] = $default_user_facility;
+
+  // Insert events - Vacation event starts at some date and repeats till end date,
+  // covering Calendar slots over entire clinic hours under selected provider.
+  InsertEvent($args);
+  refreshCalendar(); // after "Add vacation" process is complete
+}
+
+// 2. Clinic Holiday dates
+if ($_POST['form_action'] == "holiday_submit") {
+  if ($_POST['holiday_action'] == "Delete Holiday") {
+    // user requests to delete a set of holidays
+    $dateAndType = json_decode($_POST['jsonData'], true);  // convert JSON string to an associative array
+    // this associative array contains elements in form of "date" => array("day type", "facility id")
+    foreach ($dateAndType as $date => $value) {
+      // for each set of date and facility, delete holiday event under all providers
+      // no need to iterate through providers in this case
+      sqlStatement("DELETE FROM libreehr_postcalendar_events WHERE pc_eventDate = ? AND pc_facility = ?", array($date, $value[1]));
+    }
+  } elseif ($_POST['holiday_action'] == "Add Holiday") {
+    // user requests to add a set of holidays
+    // check if there is an appt. (either recurring or non-recurring) on at least one of selected holiday dates
+    $dateAndType = json_decode($_POST['jsonData'], true);  // convert JSON string to an associative array
+    // this associative array contains elements in form of "date" => array("day type", "facility id")
+    foreach ($dateAndType as $date => $value) {
+      // check each selected date
+      $fetchedEvents = fetchAllEvents($date, $date); // returns an array of events on $date or including $date (in case of recurring appts.)
+      if (!empty($fetchedEvents)) {
+        foreach ($fetchedEvents as $event) {
+          // check if event is a patient event
+          if (event['pc_pid'] > 0) {
```

Comment:
Small bug is if you are adding for a specific facility and there is an appointment in another facility you get the message. Facility checking needs to be done.

Reply:
@teryhill If a user adds a holiday for a particular date, and if an appointment is on that date even if in another facility, shouldn't it be prevented from adding so as to not overlap it?

Reply:
If one facility needs to close it should not be prevented by another facilities appointments. They are separate. You can wait until customers bring up issues if you want.

Reply:
Set up 2 facilities, add the facility to a user as the default. Add an appointment to that user's calendar. On the second facility add a holiday for the same day the appointment is on.

Reply:
@teryhill It did the alert. So, overlapping of events in case when date is same, provider is same but facility is different should be fine? And it should only alert when holiday's and appt.'s date, provider and facility are same?

Reply:
I think checking with date and facility is enough because if facility is closed, there can't be that appt. in any case, right?

Reply:
if ($event['pc_pid'] != 0) {
              if($event['pc_facility'] == $value[1]) {